### PR TITLE
how-to/maintain: refine BR doc description and organization

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? 

Refine the BR document and update v3.1

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/2084

### Which version does your change affect? 

dev, v3.1

### CODE REVIEW

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 1]
```diff
@@ -16,15 +16,15 @@ BR 是分布式备份恢复的工具，它将备份和恢复操作命令下发
 
 ## BR 命令行描述
 
-`br` 命令行是由命令（包括子命令）、选项和参数组成的。命令即不带 `-` 或者 `--` 的字符。选项即带有 `-` 或者 `--` 的字符。参数即命令或选项字符后紧跟的、传递给命令和选项的字符。
+一条 `br` 命令是由子命令、选项和参数组成的。子命令即不带 `-` 或者 `--` 的字符。选项即以 `-` 或者 `--` 开头的字符。参数即字命令或选项字符后紧跟的、并传递给命令和选项的字符。
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
一条 `br` 命令是由子命令、选项和参数组成的。子命令即不带 `-` 或者 `--` 的字符。选项即以 `-` 或者 `--` 开头的字符。参数即子命令或选项字符后紧跟的、并传递给命令和选项的字符。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 2]
```diff
@@ -52,11 +52,11 @@ BR 还包含以下三个子命令：
 * `--ca`：指定 PEM 格式的受信任 CA 的证书文件路径。
 * `--cert`：指定 PEM 格式的 SSL 证书文件路径。
 * `--key`：指定 PEM 格式的 SSL 证书密钥文件路径。
-* `--status-addr`：指定 BR 提供 Prometheus 统计的监听地址。
+* `--status-addr`：BR 向 Prometheus 提供统计数据的监听地址，
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
* `--status-addr`：BR 向 Prometheus 提供统计数据的监听地址。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 3]
```diff
@@ -75,7 +75,7 @@ br backup full \
     --log-file backupfull.log
 ```
 
-以上命令通过 `--ratelimit` 和 `--concurrency` 选项限制了 **每个 TiKV** 执行备份任务的速度上限（单位 MiB/s）和并发数上限，同时把 BR 的 log 写到 `backupfull.log` 文件中。
+以上命令中，`--ratelimit` 和 `--concurrency` 选项限制了 **每个 TiKV** 执行备份任务的速度上限（单位 MiB/s）和并发数上限。`--log-file` 选项指定把 BR 的 log 写到 `backupfull.log` 文件中。
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
以上命令中，`--ratelimit` 和 `--concurrency` 选项限制了**每个 TiKV** 执行备份任务的速度上限（单位 MiB/s）和并发数上限。`--log-file` 选项指定把 BR 的 log 写到 `backupfull.log` 文件中。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 4]
```diff
@@ -189,14 +189,14 @@ br restore table \
...
+- BR 只支持 TiDB v3.1 及以上版本。
+- 如果备份的集群没有网络存储，在恢复前需要将所有备份的 SST 文件拷贝到各个 TiKV 节点上 `--storage` 指定的目录下。
 - TiDB 执行 DDL 期间不能执行备份操作。
 - 目前不支持 Partition Table 的备份恢复。
 - 目前只支持在全新的集群上执行恢复操作。
-- 如果备份时间可能超过设定的 GC lifetime（默认 10 分钟），则需要将 GC lifetime 调大。
+- 如果备份时间可能超过设定的 [`tikv_gc_life_time`](/dev/reference/garbage-collection/configuration.md#tikv_gc_life_time)（默认 `10m0s`），则需要将该参数调大。
 
-    例如，将 GC lifetime 调整为 720 小时:
+    例如，将 `tikv_gc_life_time` 调整为 `720h`:
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
    例如，将 `tikv_gc_life_time` 调整为 `720h`：
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 5]
```diff
@@ -256,8 +259,11 @@ MySQL [sbtest]> select count(*) from sbtest1;
 |  5000000 |
 +----------+
 1 row in set (1.04 sec)
+```
 
-表结构：
+表的结构如下：
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
表结构如下：
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 6]
```diff
@@ -266,20 +272,21 @@ CREATE TABLE `sbtest1` (
...
-
 ```
 
-### 机器配置
+示例假设有 4 个 TiKV 节点，每个节点配置如下：
 
-```
-4 tikv (16 Core，32GB MEM，SSD), 3 副本
-```
+16 Core，32GB MEM，SSD, 3 副本
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
16 Core，32GB MEM，SSD，3 副本
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 7]
```diff
@@ -266,20 +272,21 @@ CREATE TABLE `sbtest1` (
...
 
-### 执行备份
+### 备份示例
 
 - 备份前需确认 GC 时间调长, 确保备份期间不会因为数据丢失导致中断
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
- 备份前需确认已将 GC 时间调长，确保备份期间不会因为数据丢失导致中断
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 8]
```diff
@@ -266,20 +272,21 @@ CREATE TABLE `sbtest1` (
...
 - 备份前需确认 TiDB 集群没有执行 DDL
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
- 备份前需确认 TiDB 集群没有执行 DDL 操作
```

**Comment by TomShawn** [comment]
@3pointer PTAL, thanks~

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 9]
```diff
@@ -279,12 +221,76 @@ br restore table \
...
+
+### 恢复原理
+
+BR 执行恢复操作时，会按顺序执行以下任务：
+
+1. 解析备份路径下的 backupMeta 文件，根据解析出来的库表信息，在内部启动一个 TiDB 实例在新集群创建对应的库表。
+
+2. 把解析出来的 SST 文件，根据表进行 `GroupBy` 聚合。
+
+3. 根据 SST 文件的 Key Range 进行预切分 Region，使得每个 Region 对应一个 SST 文件。
```
Comment (3pointer) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
3. 根据 SST 文件的 Key Range 进行预切分 Region，使得每个 Region 至少对应一个 SST 文件。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 10]
```diff
@@ -279,12 +221,76 @@ br restore table \
...
+
+4. 遍历要恢复的每一张表，以及这个表对应的 SST 文件。
+
+5. 找到该文件对应的 Region，发送下载文件的请求到对应的 TiKV 节点，并在下载成功后，发送加载请求。
+
+TiKV 收到加载 SST 文件的请求后, 利用 Raft 机制保证加载 SST 数据的强一致性。在加载成功后，下载下来的 SST 文件会被异步删除。
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
TiKV 收到加载 SST 文件的请求后，利用 Raft 机制保证加载 SST 数据的强一致性。在加载成功后，下载下来的 SST 文件会被异步删除。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 11]
```diff
@@ -241,14 +183,14 @@ br restore table \
 
 ## 注意事项
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]: How about `使用限制`?

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 12]
```diff
@@ -241,14 +183,14 @@ br restore table \
...
 
-- BR 只支持 TiDB 3.1 及以上版本。
-- 如果在没有网络存储的集群上备份，在恢复前需要将所有备份下来的 SST 文件拷贝到各个 TiKV 节点上 `--storage` 指定的目录下。
+- BR 只支持 TiDB v3.1 及以上版本。
+- 如果备份的集群没有网络存储，在恢复前需要将所有备份的 SST 文件拷贝到各个 TiKV 节点上 `--storage` 指定的目录下。
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]: Suggest moving this item to `恢复集群数据`.

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 13]
```diff
@@ -241,14 +183,14 @@ br restore table \
...
 - TiDB 执行 DDL 期间不能执行备份操作。
 - 目前不支持 Partition Table 的备份恢复。
 - 目前只支持在全新的集群上执行恢复操作。
-- 如果备份时间可能超过设定的 GC lifetime（默认 10 分钟），则需要将 GC lifetime 调大。
+- 如果备份时间可能超过设定的 [`tikv_gc_life_time`](/dev/reference/garbage-collection/configuration.md#tikv_gc_life_time)（默认 `10m0s`），则需要将该参数调大。
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]: How about moving this item to the backup section?

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 14]
```diff
@@ -241,14 +183,14 @@ br restore table \
...
 
-    例如，将 GC lifetime 调整为 720 小时:
+    例如，将 `tikv_gc_life_time` 调整为 `720h`：
 
     {{< copyable "sql" >}}
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]: @3pointer Can we take this item (为了加快数据恢复速度...) as one of the best practices section?

File: dev/how-to/maintain/backup-and-restore/br.md (hunk 11 above)
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]: How about `使用限制`?

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 15]
```diff
@@ -279,12 +217,22 @@ br restore table \
     ./pd-ctl -u ${PDIP}:2379 scheduler add balance-region-scheduler
     ```
 
-## 备份恢复事例
+## 使用限制
+
+- BR 只支持 TiDB v3.1 及以上版本。
+- TiDB 执行 DDL 期间不能执行备份操作。
+- 目前不支持 Partition Table 的备份恢复。
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
- 目前不支持分区表的备份恢复。
```

File: dev/how-to/maintain/backup-and-restore/br.md [hunk 16]
```diff
@@ -231,32 +188,13 @@ br restore table \
...
-    例如，将 GC lifetime 调整为 720 小时:
-
-    {{< copyable "sql" >}}
-
-    ```
-    mysql -h${TiDBIP} -P4000 -u${TIDB_USER} ${password_str} -Nse \
-        "update mysql.tidb set variable_value='720h' where variable_name='tikv_gc_life_time'";
-    ```
-
 - 为了加快数据恢复速度，可以在恢复操作前，使用 pd-ctl 关闭与调度相关的 schedulers。恢复完成后，再重新添加这些 schedulers。
```
Comment (lilin90) [inline dev/how-to/maintain/backup-and-restore/br.md]:
```suggestion
- 为了加快数据恢复速度，可以在恢复操作前，使用 pd-ctl 关闭与调度相关的 scheduler。恢复完成后，再重新添加这些 scheduler。
```

**Pull request merged**